### PR TITLE
feat: add C `ndarray` implementation for `blas/base/ddot`

### PR DESCRIPTION
**Pull request by aman-095**
Progresses #2039.

## Description

> What is the purpose of this pull request?

This pull request adds C `ndarray` API for `blas/base/ddot`.

## Related Issues

> Does this pull request have any related issues?

This pull request:

-   progresses #2039  

## Questions

> Any questions for reviewers of this pull request?

No.

## Other

> Any other information relevant to this pull request? This may include screenshots, references, and/or implementation notes.

No.

## Checklist

> Please ensure the following tasks are completed before submitting this pull request.

-   [x] Read, understood, and followed the [contributing guidelines][contributing].

* * *

@stdlib-js/reviewers

[contributing]: https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md

### CODE REVIEW

File: lib/node_modules/@stdlib/blas/base/ddot/README.md [hunk 1]
```diff
@@ -227,6 +227,32 @@ The function accepts the following arguments:
...
+-   **strideX**: `[in] CBLAS_INT` index increment for `X`.
+-   **offsetX**: `[in] CBLAS_INT` starting index for `X`.
+-   **Y**: `[in] double*` second input array.
+-   **strideY**: `[in] CBLAS_INT` index increment for `Y`.
+-   **offsetY**: `[in] CBLAS_INT` starting index for `Y`.
+
+```c
+double c_ddot_ndarray( const CBLAS_INT N, const double *X, const CBLAS_INT strideX, const CBLAS_INT offsetX, const double *Y, const CBLAS_INT strideY. const CBLAS_INT offsetY );
+```
+
```
Comment (kgryte) [inline lib/node_modules/@stdlib/blas/base/ddot/README.md:255]: @aman-095 Same question as `dsdot`. Did you intentionally not update the example below to match `example.c`?

**Review by kgryte — COMMENTED** [review]
Apart from the README question, this PR LGTM.

File: lib/node_modules/@stdlib/blas/base/ddot/README.md (hunk 1 above)
Comment (aman-095) [reply lib/node_modules/@stdlib/blas/base/ddot/README.md:255]: @kgryte I missed that. Updated now.

**Review by kgryte — APPROVED** [review]
LGTM. Thanks, @aman-095!

**Pull request merged**